modules/microshift-cni.adoc

```diff
@@ -0,0 +1,301 @@
+
+OVN containers::
+OVN-Kubernetes cluster manifests are in the `microshift/assets/components/ovn` directory. Two OVN-Kubernetes daemon sets are rendered and applied by {product-title}.
+
+ovnkube-master::
+Includes the `northd`, `nbdb`, `sbdb` and `ovnkube-master` containers.
+
+ovnkube-node::
+The ovnkube-node includes the OVN-Controller container. After {product-title} boots, the OVN-Kubernetes daemon sets are deployed in the `openshift-ovn-kubernetes` namespace.
+
+Packaging::
+OVN-Kubernetes manifests and startup logic are built into the {product-title} RPM. The systemd services and configurations included in `microshift-networking` RPM are:
+
+* microshift-nm.conf for NetworkManager.service
+* microshift-cpuaffinity.conf for ovs-vswitchd.service
+* microshift-cpuaffinity.conf for ovsdb-server.service
+* microshift-ovs-init.service
+* configure-ovs.sh for microshift-ovs-init.service
+* configure-ovs-microshift.sh for microshift-ovs-init.service
+
+== Configuring OVN-Kubernetes
+An OVN-Kubernetes config file must be written to /etc/microshift/ovn.yaml. {product-title} will use default OVN-Kubernetes configuration values if an OVN-Kubernetes config file is not customized.
+//Q: what are the default values?
+
+.Supported OVN-Kubernetes configurations for {product-title}.
+[cols="6",options="header"]
+|===
+|Field
+|Required
+|Type
+|Default
+|Description
+|Example
+
+|`disableOVSInit`
+|N
+|bool
+|false
+|Skip configuring OVS bridge `br-ex` in `microshift-ovs-init.service`
+|true <1>
+
+|mtu
+|N
+|uint32
+|1400
+|MTU value used for the pods
+|1300
+|===
+
+<1> The OVS bridge is required. When `disableOVSInit` is true, OVS bridge `br-ex` must be configured manually.
+
+.Example `ovn.yaml` config file:
+//Q: this doesn't match the table above; should it have all six fields?
+[source, yaml]
+----
+disableOVSInit: true
+mtu: 1300
+Configuring Host
+OVS bridge
+----
+[IMPORTANT]
+When `disableOVSInit` is set to true in the `ovn.yaml` config file, the OVS bridge br-ex must be manually configured.
+
+=== Manually configuring OVS bridge br-ex
+.Procedure
+Manually configur the OVS bridge br-ex by running the following commands.
+//Q: Need SME review of each command's function.
+* Initiate OVS:
++
+[source, terminal]
+----
+$ sudo systemctl enable openvswitch --now
+----
+
+* Add the network bridge:
++
+[source, terminal]
+----
+$ sudo ovs-vsctl add-br br-ex
+----
+
+* Add the interface to the network bridge:
++
+[source, terminal]
+----
+$ sudo ovs-vsctl add-port br-ex <physical-interface-name>
+----
+The `<physical-interface-name>` is the network interface name where the node IP address is assigned.
+
+* Get the bridge up and running:
++
+[source, terminal]
+----
+$ sudo ip link set br-ex up
+----
+//Q: Is this last sentence actually for the step preceeding.
+* After `br-ex up` is running, assign the node IP address to `br-ex` bridge:
+
+[source, terminal]
+----
+$ sudo ...
+----
```

Comment:
This procedure would not really be valid for a customer, this is acceptable for a development environment, but the problem is that this will not be persisted across reboots.

We may need to help customers define this configuration for network manager in a way that it can be installed in /etc and persisted across reboots.

@zshi-redhat ^